R/scale.R

```diff
@@ -124,11 +124,11 @@ prep.step_scale <- function(x, training, info = NULL, ...) {
 
 #' @export
 bake.step_scale <- function(object, new_data, ...) {
-  res <-
-    sweep(as.matrix(new_data[, names(object$sds)]), 2, object$sds, "/")
-  res <- tibble::as_tibble(res)
-  new_data[, names(object$sds)] <- res
-  as_tibble(new_data)
+  for (column in names(object$sds)) {
+    sd <- object$sds[column]
+    new_data[[column]] <- new_data[[column]] / sd
+  }
+  new_data
```

Comment:
I think we may still want to return `tibble::as_tibble(new_data)` in these spots unless there's some machinery that processes `new_data` beforehand and we can guarantee `new_data` is a tibble.

i.e. reading through this, I'm not sure we can guarantee we'll return a tibble—acknowledging that I see the removal of the `as.matrix` line.🙂

Reply:
Sweep isn't actually that bad, it just happens to slow us down here since we need to `as.matrix()` and `as_tibble()`.

We don't need to ensure that we get a tibble since that is done automatically after each bake step is being performed https://github.com/tidymodels/recipes/blob/77fa788b3fed542c7832fc00a8b9725a165e6a46/R/recipe.R#L580.

https://github.com/tidymodels/recipes/pull/979 I have a separate PR to remove the duplicate instances of `as_tibble()`.